src/Raven.Server/Documents/ETL/EtlProcess.cs

```diff
@@ -201,81 +194,61 @@ private IEnumerable<TExtracted> Extract(DocumentsOperationContext context, long
 
             if (Transformation.ApplyToAllDocuments)
             {
                 var docs = Database.DocumentsStorage.GetDocumentsFrom(context, fromEtag, 0, long.MaxValue).GetEnumerator();
                 scope.EnsureDispose(docs);
+                merged.AddEnumerator(ConvertDocsEnumerator(context, docs, null));
+
+                // when collection isn't specified this will get the tombstones for docs, attachments & revisions in a single enumerator
+                // otherwise we will handle attachment and documents in a dedicated enumerator
 
                 var tombstones = Database.DocumentsStorage.GetTombstonesFrom(context, fromEtag, 0, long.MaxValue).GetEnumerator();
                 scope.EnsureDispose(tombstones);
-
-                tombstones = new FilterTombstonesEnumerator(tombstones, stats, Tombstone.TombstoneType.Document, context);
-
-                enumerators.Add((docs, tombstones, null));
+                merged.AddEnumerator(ConvertTombstonesEnumerator(context, tombstones, null));
             }
             else
             {
                 foreach (var collection in Transformation.Collections)
                 {
                     var docs = Database.DocumentsStorage.GetDocumentsFrom(context, collection, fromEtag, 0, long.MaxValue).GetEnumerator();
                     scope.EnsureDispose(docs);
+                    merged.AddEnumerator(ConvertDocsEnumerator(context, docs, collection));
 
                     var tombstones = Database.DocumentsStorage.GetTombstonesFrom(context, collection, fromEtag, 0, long.MaxValue).GetEnumerator();
                     scope.EnsureDispose(tombstones);
-
-                    enumerators.Add((docs, tombstones, collection));
+                    merged.AddEnumerator(ConvertTombstonesEnumerator(context, tombstones, collection));
                 }
-            }
-
-            foreach (var en in enumerators)
-            {
-                merged.AddEnumerator(ConvertDocsEnumerator(en.Docs, en.Collection));
-                merged.AddEnumerator(ConvertTombstonesEnumerator(en.Tombstones, en.Collection, EtlItemType.Document));
-            }
-
-            if (ShouldTrackAttachmentTombstones())
-            {
-                var attachmentTombstones = Database.DocumentsStorage
-                    .GetTombstonesFrom(context, AttachmentsStorage.AttachmentsTombstones, fromEtag, 0, long.MaxValue).GetEnumerator();
-                scope.EnsureDispose(attachmentTombstones);
-
-                attachmentTombstones = new FilterTombstonesEnumerator(attachmentTombstones, stats, Tombstone.TombstoneType.Attachment, context,
-                    fromCollections: Transformation.ApplyToAllDocuments ? null : Transformation.Collections);
 
-                merged.AddEnumerator(ConvertTombstonesEnumerator(attachmentTombstones, null, EtlItemType.Document));
+                if (ShouldTrackAttachmentTombstones())
```

Comment:
You should do check and track attachments in `Transformation.ApplyToAllDocuments` as well, no?